File: gajeong/Programmers/LV2/당구연습.js
function solution(m, n, startX, startY, balls) {
  var answer = [];

  const dir = [
    [-startX, startY], //왼쪽벽 친 경우
    [2 * m - startX, startY], // 오른쪽 벽 친 경우
    [startX, 2 * n - startY], // 위쪽 벽 친 경우
    [startX, -startY], //아래 벽 친 경우
  ];

  balls.forEach((el) => {
    let [x, y] = [el[0], el[1]];
    let dmp = [];

    if (!(startX > x && startY == y))
      dmp.push((dir[0][0] - x) ** 2 + (dir[0][1] - y) ** 2);
    if (!(startX < x && startY == y))
      dmp.push((dir[1][0] - x) ** 2 + (dir[1][1] - y) ** 2);
    if (!(startY < y && startX == x))
      dmp.push((dir[2][0] - x) ** 2 + (dir[2][1] - y) ** 2);
    if (!(startY > y && startX == x))
      dmp.push((dir[3][0] - x) ** 2 + (dir[3][1] - y) ** 2);

    answer.push(Math.min(...dmp));
  });

  return answer;
}

solution(10, 10, 3, 7, [
  [7, 7],
  [2, 7],
  [7, 3],
]);

File: jiwoo/Programmers/당구 연습.cpp
#include <bits/stdc++.h>

using namespace std;

vector<int> solution(int m, int n, int startX, int startY, vector<vector<int>> balls)
{
    vector<int> answer;
    for (int k = 0; k < balls.size(); k++)
    {
        int mindist = 987654321;

        int a, b, tmp;

        if (startY != balls[k][1] || startX < balls[k][0]) // 왼쪽 벽
        {
            a = startX + balls[k][0];
            b = abs(startY - balls[k][1]);
            tmp = pow(a, 2) + pow(b, 2);

            mindist = min(tmp, mindist);
        }

        if (startX != balls[k][0] || startY < balls[k][1]) // 아래 벽
        {
            a = abs(startX - balls[k][0]);
            b = startY + balls[k][1];
            tmp = pow(a, 2) + pow(b, 2);

            mindist = min(tmp, mindist);
        }

        if (startY != balls[k][1] || startX > balls[k][0]) // 오른쪽 벽
        {
            a = (m - startX) + (m - balls[k][0]);
            b = abs(startY - balls[k][1]);
            tmp = pow(a, 2) + pow(b, 2);

            mindist = min(tmp, mindist);
        }

        if (startX != balls[k][0] || startY > balls[k][1]) // 위 벽
        {
            a = abs(startX - balls[k][0]);
            b = (n - startY) + (n - balls[k][1]);
            tmp = pow(a, 2) + pow(b, 2);

            mindist = min(tmp, mindist);
        }

        answer.push_back(mindist);
    }

    return answer;
}

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    return 0;
}